lms/djangoapps/instructor/management/commands/dump_userinfo.py

```diff
@@ -231,18 +262,41 @@ def handle(self, *args, **options):
         intervals = int(0.10 * total)
         if intervals > 100 and verbose:
             intervals = 101
+        if intervals < 1:
+            intervals = 1
 
         sys.stdout.write("Processing users")
 
         for profile in profiles:
             user_id = profile['user__id']
+            user = User.objects.get(id=user_id)
             self.print_progress(count, intervals, verbose)
 
             student_dict = {
                 'Credits Issued': 0.0,  # XXX should be revisited when credit count functionality implemented
                 'Attested': False,
             }
 
+            if include_grades:
+                # Get scores for each exam and user in a course,
+                # then create a dict that will associate the scores with the user
+                scores = self.get_scores(user, course_id)
+                exams = [
+                    {
+                        'score': grade['percent'],
+                        'label': grade['label'],
+                    }
+                    for grade in scores['section_breakdown']
+                ]
+                user_info = {
+                    'name': user,
+                    'exams': exams,
+                }
+
+                for exam in user_info['exams']:
+                    label = exam['label']
+                    student_dict[label] = exam['score']
```

Comment:
I'm worried there could be conflicts here due to key collisions.

Since this is a dictionary, If there are two scores with the same label, wouldn't the last in the list overwrite any previous values? e.g. two scores with a label "Quiz".
Or what if one of the labels matches one of the previously existing values, e.g. 'Attested`, etc.?

While possible, this wasn't previously a concern because we (the code) controlled all of the values that ended up in the `student_dict`. But now that the values are pulled dynamically from the sections, it seems like naming collisions are possible and would cause problems.

I think we may need to maintain two different data structures here:
- a dictionary, `student_dict`, could remain as it's needed for the column ordering specified by `CME_SPECIFIC_ORDER`.
- a list, `student_scores`, which is then appended to the CSV row based on the order in which they appear (but allowing for duplicate labels), similar to how it's done below with `build_new_columns`.

Thoughts?

Reply:
The class `AssignmentFormatGrader` found in `/common/lib//xmodule/xmodule/graders.py` initialize a variable named `starting_index` which according to the docstring in that class is "the first number that will appear. For example, starting_index=3 and min_count = 2 would produce the labels "Assignment 3", "Assignment 4"."

If you go to the line 363 you'll see this:

```
short_label = u"{short_label} {index:02d}".format(
                index=i + self.starting_index,
                short_label=self.short_label
            )
```
This concatenates the name of the label and a consecutive. Which means the label will never be repeated.